crates/hyperdrive-math/src/long/open.rs

```diff
@@ -357,23 +350,23 @@ mod tests {
             let id = chain.snapshot().await?;
 
             // Run the preamble.
-            let fixed_rate = fixed!(0.05e18);
-            preamble(&mut rng, &mut alice, &mut bob, &mut celine, fixed_rate).await?;
+            initialize_pool_with_random_state(&mut rng, &mut alice, &mut bob, &mut celine).await?;
 
             // Get state and trade details.
             let state = alice.get_state().await?;
+            let min_txn_amount = state.minimum_transaction_amount();
             let max_long = bob.calculate_max_long(None).await?;
-            let min_base_amount = FixedPoint::from(state.config.minimum_transaction_amount)
-                * FixedPoint::from(state.info.vault_share_price);
-            let long_amount = rng.gen_range(min_base_amount..=max_long);
+            let base_amount = rng.gen_range(min_txn_amount..=max_long);
 
             // Compare the open short call output against calculate_open_long.
-            let bonds_purchased = state.calculate_open_long(long_amount);
+            let rust_bonds = state.calculate_open_long(base_amount);
 
+            // Fund a little extra to allow for of slippage.
+            bob.fund(base_amount + fixed!(10e18)).await?;
```

Comment:
What do you think of funding extra by a percentage of the `base_amount` as opposed to a flat value?

Reply:
Good idea! I'll add it.